collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -180,13 +199,26 @@ void configureButtons(FormController formController) {
     }
 
     protected void goUpLevel() {
-        Collect.getInstance().getFormController().stepToOuterScreenEvent();
+        FormController formController = Collect.getInstance().getFormController();
+        boolean shouldShowRepeatGroupPicker = repeatGroupPickerIndex != null;
+
+        if (shouldShowRepeatGroupPicker) {
+            // Simply exit the picker.
+            repeatGroupPickerIndex = null;
+        } else {
+            // Toggle the picker if coming from an inner repeat group.
+            if (formController.getEvent(screenIndex) == FormEntryController.EVENT_REPEAT) {
```

Comment:
How does this tell you you're coming from an inner repeat? Has `screenIndex` already been set to the parent repeat? If so, please make this clear in the comment. Naively I was thinking `screenIndex` would be pointing to the group we're in before we press 'up'...

Reply:
`screenIndex` is set in `jumpToHierarchyStartIndex` which is called each time the view refreshes. So you're exactly right, it will be set to the current screen index each time before you press "go up".

I'll remove the word "inner" because that might be what's confusing here. If we're coming from **any** repeat group, this will enable the picker.